saleor/payment/utils.py

```diff
@@ -67,6 +69,30 @@
 GENERIC_TRANSACTION_ERROR = "Transaction was unsuccessful"
 ALLOWED_GATEWAY_KINDS = {choices[0] for choices in TransactionKind.CHOICES}
 
+PSP_REFERENCE_OPTIONAL_MAP = {
+    WebhookEventSyncType.TRANSACTION_REFUND_REQUESTED: [
```

Comment:
Do we need to operate on `WebhookEventSyncType`? It is different layer. IMO we should use `TransactionEventType`. 
Also, do we really need this map?  We already have the validation of the returned type inside `parse_transaction_event_data` -> `get_correct_event_types_based_on_request_type`. So we are sure, that the expected action returned a proper `event_type`. Then we don't need to have a map for optional psp_reference, as all failures and action_required are the ones which doesn't need a validation. 

This could be a simple list inside: `TransactionEventType` (We already have there a list like `FAILED_TRANSACTION_EVENTS`) like `OPTIONAL_PSP_REFERENCE_EVENTS`. Then we don't need to rely on WebhookEvent types. 
We also don't need to inject `webhook_event_type` everywhere. We already have all the details required for this action.

Reply:
Done